### ISSUE TITLE
CPU only and OSX version

### ISSUE THREAD

**Issue by hadim** [gnina/gnina#168]
Is that possible to compile gnina without CUDA for a CPU only use and also compile it for OSX?

If not, is it something that would be posssible to add?

**Comment by dkoes** [comment]
No, although I'd really like to make that possible.

**Comment by hadim** [comment]
Are all the cuda kernels also available for the CPU? If yes, I guess it's just a matter of tweaking the CMake files right?

**Comment by dkoes** [comment]
No, you need to add #ifdefs at the relevant portions of the code so there is no compile or link time dependency on CUDA.

**Comment by RMeli** [comment]
@hadim as a possible quick/temporary workaround for the time being (and only if you are using `--cnn_scoring rescore`), you could perform docking with [smina](https://sourceforge.net/projects/smina/) (setting `--num_modes` to the same value as `--num_mc_saved` to get all output poses from the MC search), and use the [gnina-torch](https://github.com/RMeli/gnina-torch) package to re-score the poses (should works on the CPU, and might work on the [GPU on Apple solicon](https://pytorch.org/blog/introducing-accelerated-pytorch-training-on-mac/)).

**Comment by XinChenQC** [comment]
Three years have passed; I hope all is well with you.

You can put the gnina binary in a Docker image with minimal CUDA runtime.

The Dockerfile should be something like,

```
FROM nvidia/cuda:12.8.1-cudnn-runtime-ubuntu24.04

COPY gnina.1.3.2 /usr/local/bin/gnina

RUN chmod +x /usr/local/bin/gnina

ENTRYPOINT ["gnina", "--no_gpu"]

```

And you pre-download the binary gnina.1.3.2 into the same file along with your Dockerfile. Build your image:

```
docker build -t my-gnina:1.3.2 .
```

And put the following container startup script into your $PATH

```
#!/bin/bash

docker run --rm -it -v "$PWD":"$PWD" -w "$PWD" my-gnina:1.3.2 "$@"

```

Finally, you can run gnina in the following convenient way.

```
gnina  -r target.pdb -l ligand.xyz --autobox_ligand ligand.xyz -o docked.sdf.gz
```

It works for my Ubuntu 24.04 without a GPU.

**Comment by dkoes** [comment]
You can also use the "official" gnina docker image: https://hub.docker.com/r/gnina/gnina

**Comment by XinChenQC** [comment]
> You can also use the "official" gnina docker image: https://hub.docker.com/r/gnina/gnina

Thanks for your reply.

I noticed this image, however is too large (27GB) for my remote computer. 

Maybe I can try a multistep build to reduce the size.
